### DISCUSSION TITLE
Is it possible to use navigation expansion only for sub-sections?

### DISCUSSION THREAD

**Discussion by StarfallProjects** [squidfunk/mkdocs-material#5051] [Q&A]
Material supports automatic navigation expansion: https://squidfunk.github.io/mkdocs-material/setup/setting-up-navigation/#with-expansion

I'd like to have the nav start out collapsed, but when a user clicks on a section, the section and all sub-sections expands.

For example:

1. User goes here: https://docs.n8n.io/integrations/ The nav appears as it is now.
2. User clicks **Creating nodes**. Instead of seeing the four headings of subsections, they see all pages listed.

I tried with partial overrides, overriding nav-item.html and changing it as follows, but encountered errors and am not sure this is the right approach anyway:

```js
{% if expandSection == true %}
   {% set expanded = true %}
{% else %}
   {% set expanded = "navigation.expand" in features %}
{% endif %}
```

Then adding expandSection:true to the frontmatter of the landing page of the section I wanted to expand.

However, I got the error `nav_item is undefined`. Interestingly, I got this error even without making any changes (just copying the existing nav-item.html into overrides)

**Comment by squidfunk** [comment]
I've not come up with an idea how this should work. Sections normally don't have pages associated with them, except for when you use `navigation.indexes` and have a `index.md` file in the same folder, so there's no way to apply metadata to a section. If you have an idea how this should work from an authors perspective in a way that doesn't feel like a hack, I'm happy to consider!

**Reply by StarfallProjects** [reply]
For my current purpose, I realised if I could just get all pages in section and render as list on the section landing page, that'd do the job. Bit of playing with the mkdocs-macros-plugin, and I got there I think. WIP here if you're curious: https://github.com/n8n-io/n8n-docs/pull/1122/files
But yeah, not really sure how to configure sections, other than basically extending MkDocs to add a section config somehow.
